src/entities/priceImpact/addLiquidityUnbalanced.ts

```diff
@@ -171,14 +170,22 @@ export const addLiquidityUnbalanced = async (
             poolTokens[tokenIndex],
             abs(deltas[tokenIndex]),
         );
-        const { bptOut: deltaBPT } = await addLiquidity.query(
-            {
-                ...input,
-                amountsIn: [absDelta.toInputAmount()],
-            },
-            poolState,
-        );
-        const signal = deltas[tokenIndex] >= 0n ? 1n : -1n;
-        return deltaBPT.amount * signal;
+        try {
+            const { bptOut: deltaBPT } = await addLiquidity.query(
+                {
+                    ...input,
+                    amountsIn: [absDelta.toInputAmount()],
+                },
+                poolState,
+            );
+            const signal = deltas[tokenIndex] >= 0n ? 1n : -1n;
+            return deltaBPT.amount * signal;
+        } catch {
+            throw new SDKError(
+                'Price Impact',
+                'Add Liquidity Unbalanced',
+                'Unexpected error while calculating addLiquidityUnbalanced PI at Delta add step',
```

Comment:
Would it be helpful for debugging if we `catch (err)` and pass it through here ?

Reply:
That was intentionally removed.
1. The SC error in this case is not so relevant to the FE (and might be misleading)
2. Removing this currently allows FE to differentiate PriceImpact errors from "pure" SC error more easily, because they rely on parsing error `message` for that - after this refactor, I'll suggest they use error `name` instead.
3. Locally for debugging it might be helpful, but I'd say it's better for us to include when we need it instead of the other way around.

Reply:
ah that makes sense thanks for explaining the context! :pray: